assets/js/ViewerApp.jsx

```diff
@@ -50,11 +57,56 @@ class ViewerApp extends Component {
     });
   }
 
+  blueLine() {
```

Comment:
Just FYI, in case you didn't know, there's a more concise way to do this:

```js
changeLine(line) {
  this.setState({ line });
}
```

and then use a lambda in the `onClick` to pass the argument:

```js
<button type="button" onClick={(e) => this.changeLine('Blue')}>
  Blue
</button>
```

However, lambdas in `render` are *usually* considered a bad practice. In fact, our linter might even complain. I know some linters do, but not sure about ours. (Maybe you had it this way originally and changed it to your current approach because of the linter?) The reasons it's considered are performance related, but for a component like this which shouldn't re-render frequently, and when it does will change the whole screen, it probably doesn't matter. So I'd suggest changing it here, to simplify the code, and have our (future) test coverage be easier unless our linter complains in which case it's okay to leave alone.

Reply:
good tip -  the linter likes it